README.md

```diff
@@ -63,19 +83,96 @@ To enable pgvectorscale:
+
+We suggest setting the index build-time paramers for major changes to index operations while query-time parameters can be used to tune the accuracy/performancy tradeoff for individual queries.
+
+ We expect most people to tune the query-time parameters (if any) and leave the index build time parameters set to default.
+
+### StreamingDiskANN index build-time parameters
+
+These parameters can be set when an index is created.
+
+| Parameter name   | Description                                                                                                                                                    | Default value |
+|------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------|
+| `storage_layout` | `memory_optimized` which uses SBQ to compress vector data or `plain` which stores data uncompressed | memory_optimized
+| `num_neighbors`    | Sets the maximum number of neighbors per node. Higher values increase accuracy but make the graph traversal slower.                                           | 50            |
+| `search_list_size` | This is the S parameter used in the greedy search algorithm used during construction. Higher values improve graph quality at the cost of slower index builds. | 100           |
+| `max_alpha`        | Is the alpha parameter in the algorithm. Higher values improve graph quality at the cost of slower index builds.                                              | 1.2           |
+| `num_dimensions` | The number of dimensions to index. By default, all dimensions are indexed. But you can also index less dimensions to make use of Matryoshka embeddings | 0 (all dimensions)
+| `num_bits_per_dimension` | Number of bits used to encode each dimension when using SBQ | 2 for less than 900 dimensions, 1 otherwise
+
+An example of how to set the `num_neighbors` parameter is:
+
+```sql
+CREATE INDEX document_embedding_idx ON document_embedding
+USING diskann (embedding) WITH(num_neighbors=50);
+```
+
+#### StreamingDiskANN query-time parameters
+
+You can also set two parameter to control the accuracy vs. query speed trade-off at query time. We suggest adjusting `diskann.query_rescore` to fine-tune accuracy.
+
+| Parameter name   | Description                                                                                                                                                    | Default value |
+|------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------|
+| `diskann.query_search_list_size` | The number of additional candidates considered during the graph search. | 100
+| `diskann.query_rescore` | The number of elements rescored (0 to disable rescoring) | 50
```

Comment:
Same comment about min/max values for the parameters

Reply:
todo later